build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/PrefixPropertyHandler.java

```diff
@@ -70,15 +71,16 @@ public void setStringValue(String key, String value) {
     }
 
     public Boolean getBooleanValue(String key, Boolean def) {
-        String value = props.get(prefix + key);
+        // First, get value using deprecated key.
+        // This check must be first, otherwise, build.gradle properties will override the CI (e.g Jenkins / teamcity) properties.
+        String value = props.get(ARTIFACTORY_PREFIX + prefix + key);
         // TODO: throw exception if not true or false. If prop set to something else
         Boolean result = (value == null) ? null : Boolean.parseBoolean(value);
         if (result != null) {
             return result;
         }
-        // Fallback, try to get value with deprecated key.
-        // This may happen if a newer version of Build-Info is used old CI which generates the deprecated properties.
-        value = props.get(ARTIFACTORY_PREFIX + prefix + key);
+        // Fallback to none deprecated key.
+        value = props.get(prefix + key);
```

Comment:
The above flow, of fetching the deprecated value, and if null, falling back to the non-deprecated value, is duplicated three times in the class. Better to take it into a separate method, and reuse it.